packages/salesforcedx-vscode-core/src/commands/forceAuthDevHub.ts

```diff
@@ -23,14 +24,18 @@ import { isDemoMode } from '../modes/demo-mode';
 
 export class ForceAuthDevHubExecutor extends SfdxCommandletExecutor<{}> {
   public build(data: {}): Command {
-    return new SfdxCommandBuilder()
-      .withDescription(
-        nls.localize('force_auth_web_login_authorize_dev_hub_text')
-      )
-      .withArg('force:auth:web:login')
+    const command = new SfdxCommandBuilder().withDescription(
+      nls.localize('force_auth_web_login_authorize_dev_hub_text')
+    );
+    if (isSFDXContainerMode()) {
+      command.withArg('force:auth:device:login');
+    } else {
+      command.withArg('force:auth:web:login');
+    }
+    command
       .withArg('--setdefaultdevhubusername')
-      .withLogName('force_auth_dev_hub')
-      .build();
+      .withLogName('force_auth_dev_hub');
```

Comment:
We need to send a different log name based on mode or we won't be able to use the metrics to see which mode was used for the command's execution.

cc @ntotten

Reply:
I'll update for both forceAuthDevHub and forceAuthWebLogin.  I'll leave the current ones unchanged, and invent 2 new ones.  

```
// Authorize a dev hub ( existing log name )
'force_auth_dev_hub'
// Authorize a dev hub ( new log name for device flow )
'force_auth_device_dev_hub'
```
and
```
// Authorize an org ( existing log name - no change )
'force_auth_web_login'
// Authorize an org via device flow ( new log name )
'force_auth_device_login'
```

Reply:
@dehru - I am going to add you to the appinsights that we have for Platform Dev Tools so you can see how this works underneath. It's easier when you are able to explore the data and see how the names affect the counts.